tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php

```diff
@@ -882,4 +882,55 @@ public function testJoinWithNonUniqueAliasThrowsException()
 
         $qb->getSQL();
     }
+
+    public function testNormalizedParamNames()
+    {
+        $qb = new QueryBuilder($this->conn);
+
+        $qb->setParameter('foo', 'foo');
+        $this->assertSame('foo', $qb->getParameter('foo'));
+        $this->assertSame('foo', $qb->getParameter(':foo'));
+
+        $qb->setParameter(':bar', 2, \PDO::PARAM_INT);
+        $this->assertSame(2, $qb->getParameter('bar'));
+        $this->assertSame(2, $qb->getParameter(':bar'));
+
+        $this->assertSame(array(
+            'foo' => 'foo',
+            'bar' => 2,
```

Comment:
> If you'd rather me just update the docs accordingly to avoid BC changes I can do that instead.

Updating [`UPGRADE.md`](https://github.com/doctrine/dbal/blob/61e1f860a304c3fb6961ecbec36927b5f20b71c6/UPGRADE.md) is sufficient